lact-gui/src/app/graphs_window/plot/imp.rs

```diff
@@ -57,17 +55,28 @@ impl WidgetImpl for Plot {
             return;
         }
 
-        let bounds = gtk::graphene::Rect::new(0.0, 0.0, width as f32, height as f32);
-        let cr = snapshot.append_cairo(&bounds);
-        // Supersample the plot area
-        let backend = CairoBackend::new(&cr, (width * 2, height * 2)).unwrap();
-        if let Err(err) = self.plot_pdf(backend) {
-            error!("Failed to plot PDF chart: {err:?}")
+        if self.dirty.replace(false) {
+            self.render_thread.replace_render_request(RenderRequest {
+                data: self.data.borrow().clone(),
+                width,
+                height,
+                title: self.title.borrow().clone(),
+                value_suffix: self.value_suffix.borrow().clone(),
+                secondary_value_suffix: self.secondary_value_suffix.borrow().clone(),
+                y_label_area_size: self.y_label_area_size.get(),
+                secondary_y_label_area_size: self.secondary_y_label_area_size.get(),
+                supersample_factor: 4,
+            });
+        }
+
+        if let Some(texture) = self.render_thread.get_last_texture() {
+            let bounds = gtk::graphene::Rect::new(0.0, 0.0, width as f32, height as f32);
+            snapshot.append_texture(&texture, &bounds);
         }
     }
 }
 
-#[derive(Default)]
+#[derive(Default, Clone)]
 #[cfg_attr(feature = "bench", derive(Clone))]
 pub struct PlotData {
     line_series: BTreeMap<String, Vec<(i64, f64)>>,
```

Comment:
Perhaps it's better to wrap the data in an `Arc<Mutex>` or `Arc<RwLock>`, so that it can be passed to the render thread without cloning the whole BTreeMap and the strings inside it (which is an expensive operation)?

Reply:
I would try that approach, it might simplify things a little bit. Would be better if all fields are wrapped around a single muted shared with render and GUI thread.

I just didn't check if gtk macros are supporting this use case.

Honestly, cost of rendering is so high, that I just don't see

Reply:
@ilya-zlobintsev On the second thought, that wouldn't work out, as we would still need to copy data, because in other case mutex lock would be held as long as frame is rendering, which isn't optimal at all.

Or we need to copy the data as soon as we acquire the mutex, which defeats the purpose.

Cloneing the data isn't that expensive to be fair in comparison to actual rendering